src/gui/sidebar/previews/layer/SidebarLayersContextMenu.cpp

```diff
@@ -0,0 +1,53 @@
+#include "gui/sidebar/previews/layer/SidebarLayersContextMenu.h"
+
+SidebarLayersContextMenu::SidebarLayersContextMenu(GladeGui* gui, SidebarToolbar* toolbar):
+        SidebarBaseContextMenu(gui->get("sidebarPreviewLayersContextMenu")) {
+
+    // Connect the context menu actions
+    const std::map<std::string, SidebarActions> ctxMenuActions = {
+            {"sidebarPreviewLayerMoveUp", SIDEBAR_ACTION_MOVE_UP},
+            {"sidebarPreviewLayerMoveDown", SIDEBAR_ACTION_MOVE_DOWN},
+            {"sidebarPreviewMergeDown", SIDEBAR_ACTION_MERGE_DOWN},
+            {"sidebarPreviewLayerDuplicate", SIDEBAR_ACTION_COPY},
+            {"sidebarPreviewLayerDelete", SIDEBAR_ACTION_DELETE},
+            // TODO: the behavior for SIDEBAR_ACTION_NEW_BEFORE/AFTER is only defined for pages
```

Comment:
See our [coding conventions](https://github.com/xournalpp/xournalpp/wiki/Coding-conventions#todo-comments)' section on `TODO`s:

```suggestion
            // TODO(ja-he): the behavior for SIDEBAR_ACTION_NEW_BEFORE/AFTER is only defined for pages
```